src/components/logger/logger.js

```diff
@@ -49,30 +50,29 @@ export default class Logger extends React.Component {
   static propTypes = {
     width: PropTypes.number.isRequired,
     height: PropTypes.number.isRequired,
-    logs: PropTypes.array.isRequired,
+    logs: PropTypes.object,
   };
 
   state = {
     events: [],
   };
 
   onClear = () => this.setState({ events: [] });
+  onEvent = (event) => this.setState((state) => ({ events: [...state.events, event] }));
 
-  componentWillUpdate({ logs }) {
-    this.logsSubscriptions.map((sub) => sub.unsubscribe());
-    this.logsSubscriptions = logs.map((obs) => {
-      return obs.subscribe((event) => {
-        this.setState((state) => ({ events: [...state.events, event] }));
-      });
-    });
+  componentWillUpdate(nextProps) {
```

Comment:
Reacts recommendation is to avoid componentWillUpdate and use componentDidUpdate instead

Reply:
And the best practice is to put this method below componentDidMount

Reply:
yes, componentDidUpdate is really great